sdk/src/types/block/semantic/state_transition.rs

```diff
@@ -327,11 +329,45 @@ impl StateTransitionVerifier for NftOutput {
 }
 
 impl StateTransitionVerifier for DelegationOutput {
-    fn creation(next_state: &Self, _context: &SemanticValidationContext<'_>) -> Result<(), StateTransitionError> {
+    fn creation(next_state: &Self, context: &SemanticValidationContext<'_>) -> Result<(), StateTransitionError> {
+        let protocol_parameters = &context.protocol_parameters;
+
         if !next_state.delegation_id().is_null() {
             return Err(StateTransitionError::NonZeroCreatedId);
         }
 
+        let slot_commitment_id = context
+            .transaction
+            .context_inputs()
+            .iter()
+            .find(|i| i.kind() == CommitmentContextInput::KIND)
+            .map(|s| s.as_commitment().slot_commitment_id())
+            .ok_or(StateTransitionError::MissingCommitmentContextInput)?;
+
+        let past_bounded_slot_index = slot_commitment_id
+            .slot_index()
+            .past_bounded_slot(protocol_parameters.max_committable_age);
+        let past_bounded_epoch_index =
+            past_bounded_slot_index.to_epoch_index(protocol_parameters.slots_per_epoch_exponent);
+
+        let registration_slot = past_bounded_epoch_index.registration_slot(
```

Comment:
Isn't it supposed to be `past_bounded_epoch_index + 1` ?

Reply:
The +1 is done inside in last_slot_index()

Reply:
Because of https://github.com/iotaledger/iota-sdk/pull/1421#discussion_r1465425423 not